File: playwrightSpecs/pages/BasePage.js
import { Header } from "./components/homePage/Header"
import { Menu } from "./components/homePage/Menu"
import { Temperature } from "./components/homePage/Temperature"


export class BasePage {    

    constructor(page){
        this.page = page
        this.header = new Header(page)
        this.menu = new Menu(page)
        this.temperature = new Temperature(page)
    }
    async navigateToHomePage(){
        await this.page.goto('/');
    }
}

File: playwrightPy/pages/BasePage.py
from playwright.sync_api import Page
from playwrightPy.pages.components.homePage.Header import Header
from playwrightPy.pages.components.homePage.Menu import Menu
from playwrightPy.pages.components.homePage.Temperature import Temperature 

class BasePage:

    def _init_(self, page: Page):
        self.page = page
        self.header = Header(page)
        self.menu = Menu(page)
        self.temperature = Temperature(page)
    
    def navigateToHomePage(self):
         self.page.goto('/')